core/src/main/java/com/graphhopper/storage/index/LocationIndex.java

```diff
@@ -62,4 +67,56 @@
     LocationIndex setApproximation(boolean approxDist);
 
     void setSegmentSize(int bytes);
+
+    /**
+     * This method explores the nodes in this LocationIndex with the specified Visitor. It guarantees to visit all
+     * unique nodes included in the queryBBox but it could visit more.
+     */
+    void query(BBox queryBBox, Visitor function);
+
+    /**
+     * This interface allows to visit every node stored in the leafs of a LocationIndex.
+     */
+    abstract class Visitor {
```

Comment:
Just as a small remark: This Visitor is not very Java-8-friendly, and also a bit unusual. If I understand it correctly, as a client, I have to override `isTileInfo` if I want `onTile` to be called? I think interfaces are most clear when control flows in one direction only: An interface is either for me to call, or for me to implement and be called.

Reply:
> An interface is either for me to call, or for me to implement and be called.

Okay, that's actually the case here. :-) But the _information_ flows both ways.

Reply:
Yes, I tried with a separate QueryOptions class but this was not much prettier. Should I revert it back to this to make it easier with lambda?

Reply:
Ah no, it's okay.

Reply:
Thanks! I'm doing more Isochrone experiments with this right now... ,-)